packages/main/src/RPA/Smartsheet.py

```diff
@@ -0,0 +1,1415 @@
+            row3 = {"rowId": 789, "column1": "value5", "column2": "value6"}
+            data = [row1, row2, row3]
+            ss.set_rows(data)
+
+            # or work with native row objects to update them.
+            row1 = ss.get_row(123)
+            for cell in row1.cells:
+                if cell.column_id == 123:
+                    cell.value = "New Value"
+            data = [row1]
+            ss.set_rows(data)
+        """  # noqa: E501
+        self._require_current_sheet()
+        new_rows = []
+        # The Table will iterate as dictionaries
+        for row in data:
+            new_rows.append(self._convert_data_to_row(row))
+        if len(new_rows) == len(self.current_sheet.rows):
+            for i, row in enumerate(self.current_sheet.rows):
+                new_rows[i].id = row.id
+        row_update_response = self.smart.Sheets.update_rows(
+            self.current_sheet.id, new_rows
+        )
+        self.logger.info(f"Updated {len(row_update_response.result)} rows.")
+        if native:
+            return row_update_response.result
+        else:
+            return [self.convert_row_to_dict(row) for row in row_update_response.result]
+
+    @keyword
+    def set_row(
+        self,
+        row: Union[int, Row],
+        data: Union[Dict, List[Dict], Row] = None,
+        native: bool = False,
+    ) -> Union[Dict, Row]:
+        """Updates a single row of the current sheet with the provided data.
+
+        You can provide the row as a native ``Row`` object or as an
+        integer representing the row ID. You may omit the ``data``
+        argument if you are providing a native ``Row`` object.
+
+        You can provide the data in several ways:
+
+        * As a dictionary: the keys of the dictionary are the
+          column IDs or Titles and the values are the values for the
+          cells.
+        * As a list of dictionaries: each dictionary is a cell. The keys
+          of the dictionary should match cell attributes, e.g., ``column_id``,
+          ``title``, ``value``, etc. See the `smartsheet API docs`_ for more
+          information. The dictionary keys must be provided in snake case.
+          You must use this method to set formulas on the row.
+        * As a native ``Row`` object: a native object from the API with
+          new values for the cells.
+
+        .. _smartsheet API docs: https://smartsheet.redoc.ly/tag/rows#operation/update-rows
+
+        For examples, see ``Set Rows``.
+        """  # noqa: E501
+        self._require_current_sheet()
+        if isinstance(row, Row) and data is None:
+            data = row
+        row_id = self._parse_row_id(row)
+        new_row = self._convert_data_to_row(data)
+        new_row.id = row_id
+        return self.set_rows([new_row], native=native)[0]
+
+    @keyword
+    def add_rows(
+        self, data: Union[List, Table], native: bool = False
+    ) -> List[Union[OrderedDict, Row]]:
```

Comment:
Similar to [this](https://github.com/robocorp/rpaframework/pull/880/files#diff-586ca41af1f200aafdbc7c787d043cc92278cfc019b8a761891814c3fc6088a0R1020).

Reply:
What I [meant](https://github.com/robocorp/rpaframework/pull/880#discussion_r1139824873).